eng/pipelines/templates/variables/globals.yml

```diff
@@ -1,12 +1,17 @@
 variables:
   DocWardenVersion: '0.7.2'
   # This is the default Java build version. It's the version used to build the shipping libraries, Spotbugs etc.
-  JavaBuildVersion: '1.17'
+  JavaBuildVersion: '1.21'
+  # This is the highest Java version we can write our Javadocs with.
+  # TODO (vcolin7): Remove and use JavaBuildVersion in templates once Javadoc migration to Java 21 is complete.
+  JavadocSafeJavaBuildVersion: '1.17'
+  # This is the highest Java version we can build Cosmos Spark connectors with.
+  JavaCosmosSparkBuildVersion: '1.17'
   # This is the default Java test version. It's the version used when running tests.
-  JavaTestVersion: '1.17'
+  JavaTestVersion: '1.21'
   # This is the latest Java version released. It's used to run a non-failing job to validate the latest Java version.
-  LatestNonLtsJavaVersion: '1.20'
-  LatestNonLtsJdkFeatureVersion: '20'
+  LatestNonLtsJavaVersion: '1.21'
+  LatestNonLtsJdkFeatureVersion: '21'
```

Comment:
This should still be 1.20 as that is the latest non-LTS version.

Reply:
I was thinking we could even get rid of these runs if we're doing everything but Javadoc generation on Java 21. Do we need to really run CI or live tests on a non-LTS version if it's not the latest version of Java?

Reply:
Yes, we should support non-LTS versions to test out new features that come out in non-LTS versions. For e.g. when Java 22 rolls out, we should be able to run tests with 22 and not wait for next LTS.